vscode/webviews/chat/cells/messageCell/human/editor/toolbar/SubmitButton.tsx

```diff
@@ -79,122 +145,257 @@ export const SubmitButton: FC<{
     onClick: (intent?: ChatMessage['intent']) => void
     isEditorFocused?: boolean
     state?: SubmitButtonState
-    className?: string
-    intent?: ChatMessage['intent']
-    onSelectIntent?: (intent: ChatMessage['intent']) => void
-}> = ({ onClick, state = 'submittable', className, intent, onSelectIntent }) => {
+    detectedIntent?: ChatMessage['intent']
+}> = ({ onClick, state = 'submittable', detectedIntent }) => {
     const experimentalOneBoxEnabled = useExperimentalOneBox()
     const {
         clientCapabilities: { agentIDE },
+        isDotComUser,
     } = useConfig()
 
-    const intentOptions = useMemo(() => getIntentOptions(agentIDE), [agentIDE])
+    const { intentDetectionDisabled, intentDetectionToggleOn, updateIntentDetectionToggle } =
+        useIntentDetectionConfig()
+
+    const intentOptions = useMemo(
+        () =>
+            getIntentOptions({
+                ide: agentIDE,
+                detectedIntent,
+                intentDetectionDisabled,
+                intentDetectionToggleOn,
+                isDotComUser,
+            }),
+        [agentIDE, detectedIntent, intentDetectionDisabled, intentDetectionToggleOn, isDotComUser]
+    )
+
+    const disabled = state === 'waitingResponseComplete' || state === 'emptyEditorValue'
 
-    if (state === 'waitingResponseComplete') {
+    const detectedIntentOption = intentOptions.find(option => option.intent === detectedIntent)
+
+    const Icon = detectedIntentOption?.intent ? detectedIntentOption.icon : Play
```

Comment:
I think this should work too

```suggestion
    const Icon = detectedIntentOption?.icon ?? Play
```

Reply:
Actually not, we have a different icon for when `intent` is `undefined` (Auto). In the dropdown it uses Sparkels but here we need to use the Play button.